### DISCUSSION TITLE
rauc 1.0: Failed to write file ?/opt/image/rauc/rootfs.1/slot.raucs.SBINAA?

### DISCUSSION THREAD

**Discussion by linuxanonimous84** [rauc/rauc#854] [Q&A]
Hello,
I'm aware that I'm using an old version of rauc, but it is what it is and I cannot change it so easily.
It has been working for a long time without problems but lately it pop up this error: 

> rauc-Message: writing data to device /dev/mmcblk0p3
> rauc-Message: installing /opt/image/systemimage.raucb: Updating slot rootfs.1 status
> rauc-Message: Updating slot file /opt/image/rauc/rootfs.1/slot.raucs
> 
> (rauc:267): rauc-WARNING **: Installation error: Error while writing status file: Failed to write file ?/opt/image/rauc/rootfs.1/slot.raucs.SBINAA?: write() failed: No space left on device
> rauc-Message: installing /opt/image/systemimage.raucb: Installation error: Error while writing status file: Failed to write file ?/opt/image/rauc/rootfs.1/slot.raucs.SBINAA?: write() failed: No space left on dee
> rauc-Message: installing /opt/image/systemimage.raucb: finished
> Installing `/opt/image/systemimage.raucb` failed
> [1970/01/01 00:04:22:6194] ERR: px2resq_install: command rauc -c /opt/image/system.conf install /opt/image/systemimage.raucb failed
> [1970/01/01 00:04:22:6196] ERR: callback_px2resq_sw_update: failed to install file

But i don't understand why it happens. 
I mean if I do a df -h i can see that the partition /opt/ is not full. 
Does anything rauc after writing the ext4 file in /dev/mmcblk0p3 inside in this partition that can lead to a full memory?
If i add the option statusfile=/tmp/global.raucs in the system.conf then doesn't fail. But make this change, can be problematic on our setup.

**Comment by ejoerns** [comment]
> Hello, I'm aware that I'm using an old version of rauc, but it is what it is and I cannot change it so easily.

You'd probably need something you can deploy updates with ;)

> It has been working for a long time without problems but lately it pop up this error:
> 
> > rauc-Message: writing data to device /dev/mmcblk0p3
> > rauc-Message: installing /opt/image/systemimage.raucb: Updating slot rootfs.1 status
> > rauc-Message: Updating slot file /opt/image/rauc/rootfs.1/slot.raucs
> > (rauc:267): rauc-WARNING **: Installation error: Error while writing status file: Failed to write file ?/opt/image/rauc/rootfs.1/slot.raucs.SBINAA?: write() failed: No space left on device
> > rauc-Message: installing /opt/image/systemimage.raucb: Installation error: Error while writing status file: Failed to write file ?/opt/image/rauc/rootfs.1/slot.raucs.SBINAA?: write() failed: No space left on dee
> > rauc-Message: installing /opt/image/systemimage.raucb: finished
> > Installing `/opt/image/systemimage.raucb` failed
> > [1970/01/01 00:04:22:6194] ERR: px2resq_install: command rauc -c /opt/image/system.conf install /opt/image/systemimage.raucb failed
> > [1970/01/01 00:04:22:6196] ERR: callback_px2resq_sw_update: failed to install file
> 
> But i don't understand why it happens. I mean if I do a df -h i can see that the partition /opt/ is not full. Does anything rauc after writing the ext4 file in /dev/mmcblk0p3 inside in this partition that can lead to a full memory? If i add the option statusfile=/tmp/global.raucs in the system.conf then doesn't fail. But make this change, can be problematic on our setup.

If you use a per-slot status file, then the status will be written into the slot you've just written. This is mounted by RAUC for this and unmounted afterwards. Thus you should not see it anymore after RAUC ran with `df -h`.

Maybe there is no space left in the file system you installed? Probably you should check this during file system image creation.

With RAUC v1..1 there was the `resize` option added to enable resizing of the ext4 after installation.


With `statusfile=/path` the status file is written to a different location, so not surprising that this should work.

**Comment by linuxanonimous84** [comment]
> If you use a per-slot status file, then the status will be written into the slot you've just written. This is mounted by RAUC for this and unmounted afterwards. Thus you should not see it anymore after RAUC ran with `df -h`.

That could explain the problem, but it is still strange. I'm going to explain my setup.
I have 2 partitions with a normal rootfs of 1 gb.
/dev/mmcblk0p2 --> Image A
/dev/mmcblk0p3 --> Image B
I have a minimal image place in other memory that it is also of 1G --> Image C.
If I do rauc install from C to A or B fails.
If i do rauc from A or B to B or A it works. If the problem is that there is no enough space on A or B to be written the status file... Why it works in this case?
Is the size of the status file, quite big?

Example of rauc from A to B 

> rauc-Message: installing /tmp/systemimage.raucb: Checking slot rootfs.0
(rauc:1402): rauc-DEBUG: Load status for slot bootloader.0.
(rauc:1402): rauc-DEBUG: Load status for slot rootfs.1.
(rauc:1402): rauc-DEBUG: Set default status for slot rootfs.0.
rauc-Message: Slot needs to be updated with /tmp/rauc/bundle/systemimage.ext4
rauc-Message: Updating /dev/mmcblk0p2 with /tmp/rauc/bundle/systemimage.ext4
rauc-Message: installing /tmp/systemimage.raucb: Updating slot rootfs.0
rauc-Message: opening slot device /dev/mmcblk0p2
rauc-Message: writing data to device /dev/mmcblk0p2
rauc-Message: installing /tmp/systemimage.raucb: Updating slot rootfs.0 status
(rauc:1402): rauc-DEBUG: Saving global slot status
rauc-Message: installing /tmp/systemimage.raucb: Updating slot rootfs.0 done

**Reply by linuxanonimous84** [reply]
Hello,
Sorry i was wrong, when we try to install a new rauc bundle from A or B, we have already set up the status file.

**Comment by linuxanonimous84** [comment]
@ejoerns  Is there any way inside the bundle, for example, pre hooks, that i can add the statusfile to be executed in the rootfs install?
